src/uWS.h

```diff
@@ -1,11 +1,30 @@
 #ifndef UWS_H
 #define UWS_H
 
+#if defined(_WIN32) && defined(__cplusplus_cli)
+#define HAVE_WIN32_CPP_CLI 1
+#pragma managed(push, off)
+#else
+#ifndef HAVE_WIN32_CPP_CLI 
+#define HAVE_WIN32_CPP_CLI 0
+#endif
+#endif
+
+#ifndef USE_WIN32_THREADS
+#if HAVE_WIN32_CPP_CLI
+#define USE_WIN32_THREADS 1
+#else
```

Comment:
Some people use ancient visual studio versions and in order to use `mutex` you need at least VS 2012. If you don't care about those (I myself certainly don't) then I'll gladly remove it :)